tests/Unit/ParallelAlgorithms/Amr/Actions/Test_CollectDataFromChildren.cpp

```diff
@@ -0,0 +1,270 @@
+std::array neighbor_3_flags{amr::Flag::DoNothing, amr::Flag::Join};
+
+DirectionMap<2, Neighbors<2>> child_1_neighbors{
+    {Direction<2>::lower_xi(),
+     Neighbors<2>{std::unordered_set{neighbor_1_id}, aligned}},
+    {Direction<2>::upper_xi(),
+     Neighbors<2>{std::unordered_set{child_2_id}, aligned}},
+    {Direction<2>::lower_eta(),
+     Neighbors<2>{std::unordered_set{child_3_id}, aligned}},
+    {Direction<2>::upper_eta(),
+     Neighbors<2>{std::unordered_set{child_3_id}, aligned}}};
+Element<2> child_1{child_1_id, std::move(child_1_neighbors)};
+
+std::unordered_map<ElementId<2>, std::array<amr::Flag, 2>>
+    child_1_neighbor_flags{{neighbor_1_id, neighbor_1_flags},
+                           {child_2_id, join_join},
+                           {child_3_id, join_join}};
+
+DirectionMap<2, Neighbors<2>> child_2_neighbors{
+    {Direction<2>::lower_xi(),
+     Neighbors<2>{std::unordered_set{child_1_id}, aligned}},
+    {Direction<2>::upper_xi(),
+     Neighbors<2>{std::unordered_set{neighbor_2_id}, rotated}},
+    {Direction<2>::lower_eta(),
+     Neighbors<2>{std::unordered_set{child_4_id}, aligned}},
+    {Direction<2>::upper_eta(),
+     Neighbors<2>{std::unordered_set{child_4_id}, aligned}}};
+Element<2> child_2{child_2_id, std::move(child_2_neighbors)};
+std::unordered_map<ElementId<2>, std::array<amr::Flag, 2>>
+    child_2_neighbor_flags{{child_1_id, join_join},
+                           {child_4_id, join_join},
+                           {neighbor_2_id, neighbor_2_flags}};
+
+DirectionMap<2, Neighbors<2>> child_3_neighbors{
+    {Direction<2>::lower_xi(),
+     Neighbors<2>{std::unordered_set{neighbor_1_id}, aligned}},
+    {Direction<2>::upper_xi(),
+     Neighbors<2>{std::unordered_set{child_4_id}, aligned}},
+    {Direction<2>::lower_eta(),
+     Neighbors<2>{std::unordered_set{child_1_id}, aligned}},
+    {Direction<2>::upper_eta(),
+     Neighbors<2>{std::unordered_set{child_1_id}, aligned}}};
+Element<2> child_3{child_3_id, std::move(child_3_neighbors)};
+std::unordered_map<ElementId<2>, std::array<amr::Flag, 2>>
+    child_3_neighbor_flags{{neighbor_1_id, neighbor_1_flags},
+                           {child_1_id, join_join},
+                           {child_4_id, join_join}};
+
+DirectionMap<2, Neighbors<2>> child_4_neighbors{
+    {Direction<2>::lower_xi(),
+     Neighbors<2>{std::unordered_set{child_3_id}, aligned}},
+    {Direction<2>::upper_xi(),
+     Neighbors<2>{std::unordered_set{neighbor_3_id}, rotated}},
+    {Direction<2>::lower_eta(),
+     Neighbors<2>{std::unordered_set{child_2_id}, aligned}},
+    {Direction<2>::upper_eta(),
+     Neighbors<2>{std::unordered_set{child_2_id}, aligned}}};
+Element<2> child_4{child_4_id, std::move(child_4_neighbors)};
+std::unordered_map<ElementId<2>, std::array<amr::Flag, 2>>
+    child_4_neighbor_flags{{child_2_id, join_join},
+                           {child_3_id, join_join},
+                           {neighbor_3_id, neighbor_3_flags}};
```

Comment:
I believe all these namespace-scope variables will be initialized every time `RunTests` is executed.  I'll let you judge whether they are expensive enough to worry about that.